### PR TITLE
Add Procurve NAC support 

### PR DESCRIPTION
**Pull request by vhuk**
Add NAC support for Procurve

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/OS/Procurve.php [hunk 1]
```diff
@@ -48,4 +52,104 @@ public function pollOS(DataStorageInterface $datastore): void
...
+        $enabled = SnmpQuery::mibs(['IEEE8021-PAE-MIB'])->hideMib()->enumStrings()->get('dot1xPaeSystemAuthControl.0')->value();
+        if ($enabled !== 'enabled') {
+            return $nac;
+        }
+
+        $rowSet = [];
+        $ifIndex_map = $this->getDevice()->ports()->pluck('port_id', 'ifIndex');
+
+        $table = SnmpQuery::mibDir('hp')->mibs(['HP-DOT1X-EXTENSIONS-MIB'])->hideMib()->enumStrings()->walk('hpicfDot1xSMAuthConfigTable')->table();
+        $table = self::pivotTable($table);
```
Comment (PipoCanaja) [inline LibreNMS/OS/Procurve.php:81]: Have you tried to use `->table(2) ` instead ? That would probably give you the correct table and allow you to remove pivotTable function.

Comment (vhuk) [reply LibreNMS/OS/Procurve.php]: Good catch, I hadn't tried that.

Refactored as suggested.

**Review by PipoCanaja — APPROVED** [review]
LGTM

**Comment by PipoCanaja** [comment]
Thx @vhuk

**Pull request merged**